extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java

```diff
@@ -61,18 +63,27 @@ public Object span(final ArcInvocationContext invocationContext) throws Exceptio
             spanContext = instrumenter.start(parentContext, methodRequest);
             scope = spanContext.makeCurrent();
         }
+        final Context currentSpanContext = spanContext;
 
         try {
             Object result = invocationContext.proceed();
-
             if (shouldStart) {
-                instrumenter.end(spanContext, methodRequest, null, null);
+                if (result instanceof Uni<?>) {
+                    ((Uni<?>) result).subscribe().with(
```

Comment:
That's not correct. 
This is going to lead to 2 subscription on the Uni/Multi - which will likely executing the logic emitting the value twice. For Multi it's even worse, as you may end up with consuming an unbounded stream without back pressure.

The solution is:

```
return uni.onTermination().invoke((item, failure, cancelled) -> instrumenter.end ....)
```

For Multi:

```
return  multi.onTermination().invoke((failure, cancellation) -> instrumenter...)
```

For Uni, if failure != null && ! cancelled -> Success. I'm not sure how cancellation should be represented in the span. Do **not** check if the item is `null`, a Uni can emit `null`.

Same for Multi (failure != null. && ! cancelled)

Reply:
Thanks. Will update the PR.